vpd-tool/src/vpd_tool.cpp

```diff
@@ -0,0 +1,74 @@
+#include "vpd_tool.hpp"
+
+#include "constants.hpp"
+#include "types.hpp"
+#include "utils.hpp"
+
+#include <iostream>
+
+namespace vpd
+{
+void VpdTool::readKeyword(const std::string& i_fruPath,
+                          const std::string& i_recordName,
+                          const std::string& i_keywordName, bool i_onHardware,
+                          const std::string& i_fileToSave)
+{
+    try
+    {
+        types::DbusVariantType l_keywordValue;
+        if (i_onHardware)
+        {
+            // TODO: uncomment when implementation of the API goes in
+            /*l_keywordValue = utils::readKeywordFromHardware(
+                constants::vpdManagerService,
+               constants::vpdManagerObjectPath,
+                constants::vpdManagerIntfName, i_fruPath,
+                make_tuple(i_recordName, i_keywordName));*/
+        }
+        else
+        {
+            std::string l_inventoryObjectPath(constants::baseInventoryPath +
+                                              i_fruPath);
+
+            l_keywordValue = utils::readDbusProperty(
+                constants::inventoryManagerService, l_inventoryObjectPath,
+                constants::ipzVpdIntf + i_recordName, i_keywordName);
+        }
+
+        if (const auto l_value =
+                std::get_if<types::BinaryVector>(&l_keywordValue))
+        {
+            const std::string& l_keywordStrValue =
+                utils::getPrintableValue(*l_value);
+
+            if (i_fileToSave.empty())
+            {
+                nlohmann::json l_resultInJson = nlohmann::json::object({});
+                nlohmann::json l_keywordValInJson = nlohmann::json::object({});
+
+                l_keywordValInJson.emplace(i_keywordName, l_keywordStrValue);
+                l_resultInJson.emplace(i_fruPath, l_keywordValInJson);
+
+                utils::printJson(l_resultInJson);
+            }
+            else
+            {
+                // TODO: Write result to a given file path.
+            }
+        }
+        else
+        {
+            std::cerr
+                << "Invalid data type received from DBus while reading keyword."
```

Comment:
when D-bus call fails to read property,  l_keywordValue will be empty.  so else case at line 59 executes.

D-bus call might fail for D-bus timeout or D-bus service unavailable or D-bus property/interface is unknown, etc., The print at line 62 is not valid.

please handle both the cases separately. 
1. D-bus read failed.
2. D-bus read is successful but unexpected data type found for the given property.

Reply:
added todo.